src/pools/D3MAaveV3Pool.sol

```diff
@@ -0,0 +1,261 @@
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+
+    function file(bytes32 what, address data) external auth {
+        require(vat.live() == 1, "D3MAaveV3Pool/no-file-during-shutdown");
+        if (what == "hub") {
+            vat.nope(hub);
+            hub = data;
+            vat.hope(data);
+        } else if (what == "king") king = data;
+        else revert("D3MAaveV3Pool/file-unrecognized-param");
+        emit File(what, data);
+    }
+
+    // Deposits Dai to Aave in exchange for adai which is received by this contract
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#deposit
+    function deposit(uint256 wad) external override onlyHub {
+        uint256 scaledPrev = adai.scaledBalanceOf(address(this));
+
+        pool.deposit(address(dai), wad, address(this), 0);
+
+        // Verify the correct amount of adai shows up
+        uint256 interestIndex = pool.getReserveNormalizedIncome(address(dai));
+        uint256 scaledAmount = _rdiv(wad, interestIndex);
+        require(adai.scaledBalanceOf(address(this)) >= (scaledPrev + scaledAmount), "D3MAaveV3Pool/incorrect-adai-balance-received");
+    }
+
+    // Withdraws Dai from Aave in exchange for adai
+    // Aave: https://docs.aave.com/developers/v/2.0/the-core-protocol/lendingpool#withdraw
+    function withdraw(uint256 wad) external override onlyHub {
+        uint256 prevDai = dai.balanceOf(msg.sender);
+
+        pool.withdraw(address(dai), wad, msg.sender);
+
+        require(dai.balanceOf(msg.sender) == prevDai + wad, "D3MAaveV3Pool/incorrect-dai-balance-received");
+    }
+
+    function exit(address dst, uint256 wad) external override onlyHub {
+        uint256 exited_ = exited;
+        exited = exited_ + wad;
+        uint256 amt = wad * assetBalance() / (D3mHubLike(hub).end().Art(ilk) - exited_);
+        require(adai.transfer(dst, amt), "D3MAaveV3Pool/transfer-failed");
+    }
+
+    function quit(address dst) external override auth {
+        require(vat.live() == 1, "D3MAaveV3Pool/no-quit-during-shutdown");
+        require(adai.transfer(dst, adai.balanceOf(address(this))), "D3MAaveV3Pool/transfer-failed");
+    }
+
+    function preDebtChange() external override {}
+
+    function postDebtChange() external override {}
+
+    // --- Balance of the underlying asset (Dai)
+    function assetBalance() public view override returns (uint256) {
+        return adai.balanceOf(address(this));
+    }
+
+    function maxDeposit() external view override returns (uint256) {
+        // Supply cap logic adapted from https://github.com/aave/aave-v3-core/blob/94e571f3a7465201881a59555314cd550ccfda57/contracts/protocol/libraries/logic/ValidationLogic.sol#L71
+        PoolLike.ReserveData memory data = pool.getReserveData(address(dai));
+        uint256 supplyCap = ((data.configuration & ~SUPPLY_CAP_MASK) >> SUPPLY_CAP_START_BIT_POSITION) * (10 ** dai.decimals());
```

Comment:
This makes the assumption that the underlying asset will always be Dai (or have the same number of decimals) btw, as long as we're OK with that

Reply:
This is an assumption of the entire D3M the variable is called `dai` :)

Reply:
Touche lol, just checking we know what we are hardcoding here